core/jobs.py

```diff
@@ -597,16 +798,44 @@ def __iter__(self):
 
     @classmethod
     def from_json(cls, input_shard_state):
+        """Creates an instance of the InputReader for the given input shard
+        state.
+
+        Args:
+            input_shard_state: dict(str : *). The InputReader state as a
+                dict-like object.
+
+        Returns:
+            *. An instance of the InputReader configured using the values of
```

Comment:
"configured using the values of json" reads weirdly to me. Perhaps "configured using the input shard state."; json is just a message format.